src/braket/circuits/circuit.py

```diff
@@ -487,6 +492,77 @@ def add_circuit(
+                Default = `None`.
+            target_mapping (dictionary[int or Qubit, int or Qubit], optional): A dictionary of
+                qubit mappings to apply to the qubits of `circuit.instructions`. Key is the qubit
+                to map, and the value is what to change it to. Default = `None`.
+
+        Returns:
+            Circuit: self
+
+        Raises:
+            TypeError: If both `target_mapping` and `target` are supplied.
+            ValueError: If `circuit` has result types attached
+
+        Examples:
+            >>> widget = Circuit().h(0).h(1)
+            >>> circ = Circuit().add_verbatim_box(widget)
+            >>> print(list(circ.instructions))
+            [Instruction('operator': StartVerbatimBlock, 'target': QubitSet([])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(0)])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(1)])),
+             Instruction('operator': EndVerbatimBlock, 'target': QubitSet([]))]
+
+            >>> widget = Circuit().h(0).cnot(0, 1)
+            >>> circ = Circuit().add_verbatim_box(widget, target_mapping={0: 10, 1: 11})
+            >>> print(list(circ.instructions))
+            [Instruction('operator': StartVerbatimBlock, 'target': QubitSet([])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(10)])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(11)])),
+             Instruction('operator': EndVerbatimBlock, 'target': QubitSet([]))]
+
+            >>> widget = Circuit().h(0).cnot(0, 1)
+            >>> circ = Circuit().add_verbatim_box(widget, target=[10, 11])
+            >>> print(list(circ.instructions))
+            [Instruction('operator': StartVerbatimBlock, 'target': QubitSet([])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(10)])),
+             Instruction('operator': H('qubit_count': 1), 'target': QubitSet([Qubit(11)])),
+             Instruction('operator': EndVerbatimBlock, 'target': QubitSet([]))]
+        """
+        if target_mapping and target is not None:
+            raise TypeError("Only one of 'target_mapping' or 'target' can be supplied.")
+        elif target is not None:
+            keys = sorted(verbatim_circuit.qubits)
+            values = target
+            target_mapping = dict(zip(keys, values))
+
+        if verbatim_circuit.result_types:
+            raise ValueError("Verbatim subcircuit is not measured and cannot have result types")
+
+        if verbatim_circuit.instructions:
+            self.add_instruction(Instruction(compiler_directives.StartVerbatimBox()))
+            for instruction in verbatim_circuit.instructions:
+                self.add_instruction(instruction, target_mapping=target_mapping)
+            self.add_instruction(Instruction(compiler_directives.EndVerbatimBox()))
+
+        self._has_compiler_directives = True
```

Comment:
If verbatim_circuit is empty then should self._has_compiler_directives be True ?

Reply:
Fixed